lib/internal/assert/assertion_error.js

```diff
@@ -47,6 +47,15 @@ function copyError(source) {
     __proto__: null,
     value: source.message,
   });
+  if (source.cause !== undefined) {
+    let cause = source.cause;
+
+    if (cause instanceof Error) {
```

Comment:
Nit: Using `util.types.isNativeError(...)` might be slightly more efficient here and should catch errors created under other contexts. Soon this can likely be replaced with `Error.isError(...)` once that is generally available in v8. https://tc39.es/proposal-is-error/